### PR TITLE
[GuestConfiguration] GA/stable version specs

### PR DESCRIPTION
**Pull request by viveklingaiah**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

**Comment by vivlingaiah** [comment]
Hi, this is after copying the specs from "Preview" to "stable" using aka.ms/openapiportal

**Comment by vivlingaiah** [comment]
please review the latest commit. the previous commits on this PR is from aka.ms/openapiportal tool when I moved the specs from preview to stable

**Comment by anuchandy** [comment]
@vivlingaiah is this api-version already deployed to any production region? Asking because merging a spec into public master requires this.

**Comment by anuchandy** [comment]
Talked to @vivlingaiah, the service is yet to be deployed, planned deployment date is 11/13/2018.

**Comment by anuchandy** [comment]
@KrisBash, @ravbhatnagar Please review from ARM side. This stable api-version is same as the preview api-version that is approved by ARM. But still requires ARM review given this is a new api-version.

**Comment by ravbhatnagar** [comment]
@vivlingaiah - Since its a new API version, lets schedule a meeting to close the review. ALso, there is a file which got added which says not implemented, not sure what that is about.

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        "flow": "implicit",
+        "description": "Azure Active Directory OAuth2 Flow",
+        "scopes": {
+          "user_impersonation": "impersonate your user account"
+        }
+      }
+    },
+    "paths": {
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}": {
+        "delete": {
```
Comment (KrisBash) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json]: Please plan to implement delete as soon as possible. A resource that can be PUT but not deleted cannot be cleaned up

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state, which only appears in the response.",
+          "x-nullable": true,
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          },
+          "enum": [
```
Comment (KrisBash) [inline specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json]: With only "Succeeded" as a value, it looks like provisioningState is spurious.

**Review by KrisBash — APPROVED** [review]
Signing off with comment. Please review comments

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json (hunk 1 above)
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json]: We will do. Esha has started to work on it.

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/VmNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
[the remark this answers is not in the document]
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json]: done

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json [hunk 4]
```diff
@@ -0,0 +1,784 @@
...
+        }, 
+        "name": {
+          "type": "string",
+          "description": "Name of the guest configuration."
+        },
+        "version": {
+          "type": "string",
+          "description": "Version of the guest configuration."
+        },
+        "contentUri": {
```
[the remark this answers is not in the document]
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json]: I checked with Amit and Esha. This is Esha needed for Custom policies in future. Custom policy is being worked on by Amit right now. So, will retain.

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json [hunk 5]
```diff
@@ -0,0 +1,784 @@
...
+            "modelAsString": true
+          },
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Pending"
+          ]
+        },
+        "lastComplianceStatusChecked": {
+          "type": "string",
```
[the remark this answers is not in the document]
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json]: done

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json (hunk 2 above)
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration.json]: added other enum values

File: specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json (hunk 1 above)
Comment (vivlingaiah) [reply specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2018-11-20/guestconfiguration_NotImplemented.json]: Added specs for delete now. Esha is about to deploy the delete API as well ?

**Comment by vivlingaiah** [comment]
Re-added/moved specs for delete guestConfigurationAssignment as it is implemented now. About to be deployed

**Comment by anuchandy** [comment]
@vivlingaiah please resolve the conflicts in readme file

**Review by anuchandy — APPROVED** [review]

**Pull request merged**